klyshko-operator/controllers/tuplegenerationtask_controller.go

```diff
@@ -233,6 +234,10 @@ func (r *TupleGenerationTaskReconciler) Reconcile(ctx context.Context, req ctrl.
 				Requeue: true,
 			}, r.setState(ctx, *taskKey, status, klyshkov1alpha1.TaskFailed)
 		}
+	case klyshkov1alpha1.TaskFailed, klyshkov1alpha1.TaskCompleted:
```

Comment:
As this statement completes the list of all defined states in [tuplegenerationtask_types.go](https://github.com/carbynestack/klyshko/blob/b12c4022f692c809eaef83f091a4c1e397e41a35/klyshko-operator/api/v1alpha1/tuplegenerationtask_types.go#L21) (`TaskPreparing`, `TaskLaunching`, `TaskGenerating`, `TaskProvisioning`, `TaskCompleted` and `TaskFailed`) with all cases returning, it seems like the log statement below can no longer be reached but for undefined states. The latter should probably be handled in a default statement and logged  / handled as an error.

Reply:
I was expecting the final statement to be reached because of the previous Requeue, however it is true it would have been not accurate to call a failed Task a "desired state".

We are now treating the unexpected state for the Task CRD explicitly. No PVC deletion in case it is needed for debugging. I considered panic-ing but couldn't bring myself to it.

Let me know if you were thinking of handling it in some other way.

Reply:
Looks good to me. 
Unexpected states are all covered by the `default` case and I cannot think of anything else than writing an error log on a panic recovery anyways.  I wouldn't handle it differently until we find an explicit reason to do so.